### DISCUSSION TITLE
Configurations files with symlink or not?

### DISCUSSION THREAD

**Discussion by Guilouz** [Guilouz/Creality-Helper-Script-Wiki#534] [Polls]
In view of the next update of the script, I ask you whether you prefer configuration files symlinked or not.

**Configuration files with symlink:**

- When updating the script, configuration files are updated automatically
- No need to uninstall and reinstall the features to benefit from the changes
- Configuration files cannot be modified from the Web interface

**Configuration files without symlink:**

- When updating the script, configuration files are not updated automatically
- Need to uninstall and reinstall the features to benefit from the changes
- Configuration files can be modified from the Web interface

**Comment by spitfire** [comment]
THe more I think about it the less confident I am, but for ease of use symlinks are better (I guess?).

**Comment by Nastras** [comment]
I personally don't like the symlinks because all individual changes are overwritten. Wouldn't it be possible for the script to ask whether it installs with or without?

**Reply by Guilouz** [reply]
I have do not plan to do this (a lot of changes in the code for so little).

**Reply by Nastras** [reply]
That's almost what I thought.
Then I still have to overwrite the symlinks with the custom.cfg file after installation 😅

**Reply by spitfire** [reply]
I’ve created my own cfg file which I’ve included in printer.cfg after the ones created by script, and this seems to be the best way for me.

**Comment by Anonoutlaw187** [comment]
still kind of new at all this printing stuff. and I removed a script from the file and had to reset the printer and start back over. wish it could pickup what was taken out and reinstall it. tried to go back in and remove it and reinstall it and man that was a mess cause it said it wasn't there and so I went in to install it and it said it was installed already. 

new idea if you can make it wo where it can pick up what was taken out and update the file from there. maybe idk 
like I said I'm new this this but I didn't like been stuck on creality print or tied to them as it is a open source printer. and I'm having a problem where it keeps kicking active bed leveling out. I have the k1c 

my vote is to make it easy.. of and maybe a server for the people that's using the install helper

**Comment by Ltek** [comment]
@Guilouz  Option 3 : automate script feature reinstall.   Have a HelperReinstall.CFG which holds a list of script features the user wants automatically reinstalled... your script reads the HelperReinstall.CFG and automatically reinstalls the items when a 'reinstall from cfg' options in the menu is selected. This cfg is automatically created/updated each time the user installs (or uninstalls) script features.

If a user has additional settings I'd suggest they have a separate config file they 'include' in printer.cfg

**Comment by NexGen-3D-Printing** [comment]
I understand why these scripts are automated, as this has to be the most complicated Klipper layout, I dont like them being over written, but it maybe for the better that they are, all of the machines I build, are far easier to manage and much cleaner and easier to access, alter and maintain, so I will probably accept having everything over written, and I will just over wright everything myself afterwards, as I have already heavily modified things like active internal air filtration, preheat sequences and various other adjustments.

**Reply by spitfire** [reply]
You can still override them easily as needed if you create your own file with macros and include it in printer.cfg after the ones from this script. I think it's the best of both worlds
